mllib/src/main/scala/org/apache/spark/ml/feature/RFormula.scala

```diff
@@ -362,6 +381,16 @@ class RFormulaModel private[feature](
         case _ => true
       }
       StructType(withFeatures.fields :+ StructField($(labelCol), DoubleType, nullable))
+    } else if (resolvedFormula.evalExprs.contains(resolvedFormula.label)) {
+      val spark = SparkSession.builder().getOrCreate()
+      val dummyRDD = spark.sparkContext.parallelize(Seq(Row.empty))
+      val dummyDF = spark.createDataFrame(dummyRDD, schema)
+        .withColumn(resolvedFormula.label, expr(resolvedFormula.label))
+      val nullable = dummyDF.schema(resolvedFormula.label).dataType match {
```

Comment:
hmm, I'm not sure about this - it's creating a session (referencing the default anyway) and then making a RDD on an empty Row? is it no other way to do this?

Reply:
In order to create a schema with arbitrary types, I actually borrowed this pattern from here: https://github.com/apache/spark/blob/master/mllib/src/main/scala/org/apache/spark/ml/feature/SQLTransformer.scala#L79

However, the datatype check to determine whether the field should be nullable seems unnecessary here, because only `NumericType` and `BooleanType` seems to be supported, so I modified here to return schema without checking datatype. https://github.com/apache/spark/blob/master/mllib/src/main/scala/org/apache/spark/ml/feature/RFormula.scala#L390

Reply:
I'm resolving this as this part got removed due to the suggested changes.